### PR TITLE
feat(frontend): add link to download QR-Code

### LINKED ISSUE #2825: 🐛 [Bug] Frontend: On Safari we can not interact with the QR-Code picture
## 🐛 Bugreport

As user if I go on the frontend and generate a link, I can not interact with the QR-Code picture.
Expected would be that I can use 'save', 'save under...' and other actions on the picture

## 🤖 Todo
- [x] Analyze & fix

### PR DESCRIPTION
**Pull request by Elweyn**
### Issues

- fixes #2825

### Todo

- [X] Add link to download QR-Code
- [x] Make unit test for add method
- [ ] Bernd needs to decide if rename 'Herunterladen' in 'Bild herunterladen', 'Bild speichern', 'QR-Code als Bild speichern'

### CODE REVIEW

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]
image

layouting is suboptimal

Suggestion: Consider to rename `Herunterladen` in `Bild herunterladen`, `Bild speichern`, `QR-Code als Bild speichern`

File: frontend/src/components/QrCode/FigureQrCode.vue [hunk 1]
```diff
@@ -37,6 +42,13 @@ export default {
       }
     }
   },
+  methods: {
+    downloadImg() {
+      const canvas = document.getElementById('qrcanvas')
+      const image = canvas.toDataURL('image/png')
+      document.getElementById('download').href = image
+    },
+  },
```
Comment (mahula) [inline frontend/src/components/QrCode/FigureQrCode.vue:51]: What about testing this method?

qr_tests

**Review by mahula — CHANGES_REQUESTED** [review]

File: frontend/src/components/QrCode/FigureQrCode.vue (hunk 1 above)
Comment (mahula) [reply frontend/src/components/QrCode/FigureQrCode.vue]: Resolved by last two commits.

**Review by mahula — APPROVED** [review]
Good solution.
I approve.
Please ping Bernd again to comment on the link naming.

**Pull request merged**